ddtrace/tracer/span_test.go

```diff
@@ -204,6 +204,27 @@ func TestSpanFinishWithErrorStackFrames(t *testing.T) {
 	assert.Equal(strings.Count(span.Meta[ext.ErrorStack], "\n\t"), 2)
 }
 
+// nilStringer is used to test nil detection when setting tags.
+type nilStringer struct {
```

Comment:
Won't `var nilStringer fmt.Stringer = nil` work equally well? We could likely drop the struct.